dask/dataframe/io.py

```diff
@@ -29,6 +29,20 @@
 csv_defaults = {'compression': None}
 
 
+def pandas_empty_like(x):
+    """Return an empty pandas dataframe or series with the same columns/name
+    and dtype. With an index with the same name and dtype"""
+    in_index = x.index
+    index = pd.Index(range(0), dtype=in_index.dtype, name=in_index.name)
+    dtypes = x.dtypes
+    if isinstance(x, pd.DataFrame):
+        columns = x.columns
+        out_dict = dict((c, np.empty(0, d)) for c, d in zip(columns, dtypes))
+        return pd.DataFrame(out_dict, index=index)
+    elif isinstance(x, pd.Series):
+        return pd.Series({x.name: np.empty(dtypes)}, index=index)
```

Comment:
We might also not want to combine this all into one function.

it would also need  to support index objects.